src/components/Delegates/DelegateCard/PartialDelegateButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { Button } from "@/components/Button";
+import { useOpenDialog } from "@/components/Dialogs/DialogProvider/DialogProvider";
+import { DelegateChunk } from "@/app/api/common/delegates/delegate";
+import { fetchCurrentDelegatees } from "@/app/delegates/actions";
+
+interface Props {
+  full: boolean;
+  delegate: DelegateChunk;
+}
+
+export const PartialDelegateButton = ({ full, delegate }: Props) => {
+  const openDialog = useOpenDialog();
+
+  return (
+    <Button
+      onClick={(e: any) => {
+        e.preventDefault();
+        openDialog({
+          type: "PARTIAL_DELEGATE",
+          params: {
+            delegate,
+            fetchCurrentDelegatees,
```

Comment:
man, this is one of my least favorite things about server components. I'm guessing we cannot just import fetchCurrentDelegates from the modal itself, because the modal is a client component and we can't import server actions in a client component, is this correct? So instead we have to import the server action higher up, then pass it down via props. It's a confusing pattern for me. I feel like the best devx or ergonomics would be being able to write the component, and import the server action where you need it. 

Not a comment on your code, maybe there's nothing we can do about it. But if there is a way for us to solve this pattern a bit better, I think it would be awesome. Maybe it's impossible though

Reply:
Completely agree. This trips me up every time. Maybe we can start wrapping some of these ss calls into hooks?